src/JSInterop/Microsoft.JSInterop/src/JSRuntime.cs

```diff
@@ -53,6 +56,12 @@ protected JSRuntime()
         /// </summary>
         protected TimeSpan? DefaultAsyncTimeout { get; set; }
 
+        /// <summary>
+        /// Stores the maximum SignalR message size accepted.
+        /// Used to restrict the total byte array transfers of a single call.
+        /// </summary>
+        protected long SignalRMaxMessageSizeBytes { get; set; } = 31 * 1024;
```

Comment:
@SteveSandersonMS See my comment below. I think this needs to be internal in RemoteJSRuntime and that we need to make `ReceiveBytes` virtual and make that check in an override within RemoteJSRuntime. That way Webassembly or other platforms don't need to know anything about this

Reply:
I think that's also difficult from a responsibility-separating point of view, as it means `RemoteJSRuntime` is now responsible for keeping track of when the data is consumed as well as when it's received.

To me it seems simpler and totally satisfactory to have a concept of a "max bytes recieved" that's platform-agnostic (and can default to `long.MaxValue`).

Reply:
> I think that's also difficult from a responsibility-separating point of view, as it means `RemoteJSRuntime` is now responsible for keeping track of when the data is consumed as well as when it's received.

Not really?

I was imagining something like this on RemoteJSRuntime
```csharp
public override void ReceiveBytes(int id, byte [] data)
{
   if(id == 0) 
   {
       ReceivedBytesCount = 0;
   }

   if(ReceivedBytes + data.Length > MaxSignalrMessageSizeInBytes)
   {
       throw new InvalidOperationException("TooMuchData");
   }
   base.ReceiveBytes(int id, byte [] data);
}
```

The reason I think this is better is because it avoids additional public API surface. We've normally pushed these concepts/concerns down to the Blazor Server layer, however I'm fine if you are ok with a general purpose property and a default limit that effectively disables it. There's really not a way for anyone to set this other than producing their own implementation

Reply:
> I was imagining something like this on RemoteJSRuntime

Very good point. I hadn't thought of resetting purely based on the `id` but yes as far as I can tell that should be safe and legal (the caller can't give us more than one `id=0` value without making us discard the underlying storage anyway).

So yes, your approach looks best to me.